install.sh

```diff
@@ -90,162 +142,396 @@ detectProfile() {
 }
 
-# Start a interactive session, asks for all the varibles, exit if running in a non-tty
-startInteractive() {
-    printf "Interactive Mode\n"
-    printf "%s\n" "Press return for default values.."
+###################################################
+# Start a interactive session, asks for all the varibles.
+# Globals: 1 variable, 1 function
+#   Variable - __VALUES_ARRAY ( array )
+#   Function - _clear_line
+# Arguments: None
+# Result: read description
+#   If tty absent, then exit
+###################################################
+_start_interactive() {
+    _print_center "justify" "Interactive Mode" "="
+    _print_center "justify" "Press return for default values.." "-"
     for i in "${__VALUES_ARRAY[@]}"; do
         j="${!i}" && k="${i}"
         read -r -p "${i} [ Default: ${j} ]: " "${i?}"
         if [[ -z ${!i} ]]; then
             read -r "${k?}" <<< "${j}"
         fi
     done
-    for _ in "${__VALUES_ARRAY[@]}"; do clearLine 1; done
-    for _ in {1..3}; do clearLine 1; done
+    for _ in "${__VALUES_ARRAY[@]}"; do _clear_line 1; done
+    for _ in {1..3}; do _clear_line 1; done
     for i in "${__VALUES_ARRAY[@]}"; do
         if [[ -n ${i} ]]; then
             printf "%s\n" "${i}: ${!i}"
         fi
     done
 }
 
+###################################################
 # Install the script
-install() {
+# Globals: 10 variables, 6 functions
+#   Variables - INSTALL_PATH, INFO_PATH, UTILS_FILE, COMMAND_NAME, SHELL_RC, CONFIG,
+#               TYPE, TYPE_VALUE, REPO, __VALUES_ARRAY ( array )
+#   Functions - _print_center, _newline, _clear_line
+#               _get_latest_sha, curl --compressed, _update_config
+# Arguments: None
+# Result: read description
+#   If cannot download, then print message and exit
+###################################################
+_install() {
     mkdir -p "${INSTALL_PATH}"
-    printf 'Installing google-drive-upload..\n'
-    printf "Fetching latest sha..\n"
-    LATEST_CURRENT_SHA="$(getLatestSHA "${TYPE}" "${TYPE_VALUE}" "${REPO}")"
-    clearLine 1
-    printf "Latest sha fetched\n" && printf "Downloading script..\n"
-    if curl -Ls --compressed "https://raw.githubusercontent.com/${REPO}/${LATEST_CURRENT_SHA}/upload.sh" -o "${INSTALL_PATH}/${COMMAND_NAME}"; then
+    _print_center "justify" 'Installing google-drive-upload..' "-"
+    _print_center "justify" "Fetching latest sha.." "-"
+    LATEST_CURRENT_SHA="$(_get_latest_sha "${TYPE}" "${TYPE_VALUE}" "${REPO}")"
+    _clear_line 1
+    _print_center "justify" "Latest sha fetched." "=" && _print_center "justify" "Downloading script.." "-"
+    if curl --compressed -Ls "https://raw.githubusercontent.com/${REPO}/${LATEST_CURRENT_SHA}/${UTILS_FILE}" -o "${INSTALL_PATH}/${UTILS_FILE}" &&
+        curl --compressed -Ls "https://raw.githubusercontent.com/${REPO}/${LATEST_CURRENT_SHA}/upload.sh" -o "${INSTALL_PATH}/${COMMAND_NAME}"; then
+        sed -i "2a UTILS_FILE="${INSTALL_PATH}/${UTILS_FILE}"" "${INSTALL_PATH}/${COMMAND_NAME}"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5557339)